hq/app/aws/inspector/Inspector.scala

```diff
@@ -44,14 +31,19 @@ object Inspector {
     }
   }
 
-  def allInspectorRuns(accounts: List[AwsAccount])(implicit ec: ExecutionContext): Attempt[List[(AwsAccount, Either[FailedAttempt, List[InspectorAssessmentRun]])]] = {
-    Attempt.labelledTaverseWithFailures(accounts)(Inspector.inspectorRuns)
+  def allInspectorRuns(accounts: List[AwsAccount], inspectorClients: Map[(String, Regions), AmazonInspectorAsync])(implicit ec: ExecutionContext): Attempt[List[(AwsAccount, Either[FailedAttempt, List[InspectorAssessmentRun]])]] = {
+    Attempt.labelledTraverseWithFailures(accounts)(Inspector.inspectorRuns(inspectorClients))
   }
 
-  def inspectorRuns(account: AwsAccount)(implicit ec: ExecutionContext): Attempt[List[InspectorAssessmentRun]] = {
+  def inspectorRuns(inspectorClients: Map[(String, Regions), AmazonInspectorAsync])(account: AwsAccount)(implicit ec: ExecutionContext): Attempt[List[InspectorAssessmentRun]] = {
     val region = Regions.EU_WEST_1  // we only automatically run inspections in Ireland
-    val inspectorClient = client(account, region)
+
     for {
+      inspectorClient <- Attempt.fromOption(inspectorClients.get((account.id, region)), FailedAttempt(Failure(
```

Comment:
Why not a `clients` function like we have for the other services?

Reply:
(I generally try to avoid implementation in a for-comp, which is a good rule of thumb for leading to code that separates logic from IO)

Reply:
Done.